src/components/regions/RegionalInsightsPanel.tsx

```diff
@@ -0,0 +1,160 @@
+import { t } from "i18next";
+import { getSortedEntityKPIValues } from "@/utils/data/sorting";
+import KPIDetailsPanel from "../ranked/KPIDetailsPanel";
+import InsightsList from "../ranked/InsightsList";
+import { Region } from "@/types/region";
+import { KPIValue } from "@/types/entity-rankings";
+
+interface InsightsPanelProps {
+  regionData: Region[];
+  selectedKPI: KPIValue;
+}
+
+function RegionalInsightsPanel({
+  regionData,
+  selectedKPI,
+}: InsightsPanelProps) {
+  if (!regionData?.length) {
+    return (
+      <div className="bg-white/5 backdrop-blur-sm rounded-level-2 p-8 h-full flex items-center justify-center">
+        <p className="text-white text-lg">{t("noData")}</p>
+      </div>
+    );
+  }
+
+  const validData = regionData.filter((m) => {
+    const value = m[selectedKPI.key as keyof Region];
+
+    // Handle boolean values if KPI is binary
+    if (selectedKPI.isBoolean) {
+      return typeof value === "boolean";
+    }
+
+    return typeof value === "number" && !isNaN(value as number);
+  });
+
+  if (!validData.length) {
+    return (
+      <div className="bg-white/5 backdrop-blur-sm rounded-level-2 p-8 h-full flex items-center justify-center">
+        <p className="text-white text-lg">
+          {t("noData", {
+            metric: selectedKPI.label,
+          })}
+        </p>
+      </div>
+    );
+  }
+
+  const sortedData = getSortedEntityKPIValues(regionData, selectedKPI);
+
+  const topMunicipalities = sortedData.slice(0, 5);
+  const bottomMunicipalities = sortedData.slice(-5).reverse();
+
+  const values = validData.map((m) => {
+    const value = m[selectedKPI.key as keyof Region];
+    // Convert boolean to number for calculations if KPI is binary
+    if (selectedKPI.isBoolean && typeof value === "boolean") {
+      return value ? 1 : 0;
+    }
+    return value as number;
+  });
+  const average = values.reduce((sum, val) => sum + val, 0) / values.length;
+
+  // For boolean KPIs, these counts have a different meaning
```

Comment:
wouldn't this be the same for both regions and municipalities? I think we can extract this as a util maybe that both can use